src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -40,23 +40,34 @@ public FindCommand parse(String args) throws ParseException {
 
         argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS);
 
-        NameContainsSubstringPredicate namePredicate = new NameContainsSubstringPredicate(
-                ParserUtil.parseSearchString(argMultimap.getValue(PREFIX_NAME).orElse("")));
-        PhoneContainsSubstringPredicate phonePredicate = new PhoneContainsSubstringPredicate(
-                ParserUtil.parseSearchString(argMultimap.getValue(PREFIX_PHONE).orElse("")));
-        EmailContainsSubstringPredicate emailPredicate = new EmailContainsSubstringPredicate(
-                ParserUtil.parseSearchString(argMultimap.getValue(PREFIX_EMAIL).orElse("")));
-        AddressContainsSubstringPredicate addressPredicate = new AddressContainsSubstringPredicate(
-                ParserUtil.parseSearchString(argMultimap.getValue(PREFIX_ADDRESS).orElse("")));
-        NoteContainsSubstringPredicate notePredicate = new NoteContainsSubstringPredicate(ParserUtil
-                .parseSearchString(argMultimap.getValue(PREFIX_NOTE).orElse("")));
-        TagSetContainsAllTagsPredicate tagsPredicate = new TagSetContainsAllTagsPredicate(
-                ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG)));
-
-        CombinedPredicates predicates = new CombinedPredicates(namePredicate, phonePredicate, emailPredicate,
-                addressPredicate, notePredicate, tagsPredicate);
-
-        return new FindCommand(predicates);
+        // Check if no attribute is provided e.g (find john)
+        if (!argMultimap.getPreamble().isEmpty() && argMultimap.getValue(PREFIX_NAME).isEmpty()
+            && argMultimap.getValue(PREFIX_PHONE).isEmpty() && argMultimap.getValue(PREFIX_EMAIL).isEmpty()
+                && argMultimap.getValue(PREFIX_ADDRESS).isEmpty() && argMultimap.getValue(PREFIX_NOTE).isEmpty()
+                && argMultimap.getAllValues(PREFIX_TAG).isEmpty()) {
```

Comment:
Is there a more elegant way to do this?

Perhaps a `noAttributesSpecified(): boolean` method in `ArgumentMultimap`?

Side note: `ArgumentMultimap` contains a `contains(Prefix): boolean` method that checks if a certain prefix has been provided, which seems like what you were trying to accomplish with the all the `getValue.isEmpty()` checks. Was there a reason why `contains()` couldn't be used?

Reply:
i overlook that contains method, thanks for pointing it out, have made the changes